pageserver/src/tenant_mgr.rs

```diff
@@ -396,9 +307,14 @@ pub async fn delete_timeline(tenant_id: ZTenantId, timeline_id: ZTimelineId) ->
     info!("waiting for timeline tasks to shutdown");
     task_mgr::shutdown_tasks(None, Some(tenant_id), Some(timeline_id)).await;
     info!("timeline task shutdown completed");
-    match tenants_state::read_tenants().get(&tenant_id) {
-        Some(tenant) => tenant.repo.delete_timeline(timeline_id)?,
-        None => anyhow::bail!("Tenant {tenant_id} not found in local tenant state"),
+    match get_tenant(tenant_id, true) {
+        Ok(tenant) => {
+            tenant.delete_timeline(timeline_id)?;
+            if tenant.list_timelines().is_empty() {
```

Comment:
> but in doc comments we typically say that tenant without a timelines is rather an anomaly and can be cause only by errors, so in other words it is not a normal situation

I don't think this way, due to all our work on splitting tenant and timeline creation: now they are created separately, so it's completely normal to have an empty tenant.
We also don't anyhow sync the tenant config (albeit, not changing either), so there's some sense to keep the tenant even there're no timelines in it: config update operations are still valid.

So, not sure it's such an abnormal situation, we might revisit it again.

Reply:
> We also don't anyhow sync the tenant config

Thats the way to avoid unnecessary syncing, no tenant no config :)

Anyway, thats out of scope for PR, this requirement should come from the control plane side, so storage will follow